enumerables.rb

```diff
@@ -0,0 +1,145 @@
+module Enumerable
+  def my_each
+    count = 0
+    while count < length
+      yield(self[count])
+      count += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    count = 0
+    while count < length
+      yield(self[count], count)
+      count += 1
+    end
+    self
+  end
+
+  def my_select
+    result = []
+    my_each do |item|
+      result << item if yield(item)
+    end
+    result
+  end
+
+  def my_all?
+    result = true
+    my_each do |item|
+      result = false unless yield(item)
+    end
+    result
+  end
```

Comment:
- [ ] should have working `my_all?` *identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- Your method doesn't consider the possibility when no block or argument is given. It should return true if the enumerable contains truthy values and false if it contains at least one falsy value.

- Your method doesn't consider the possibility when a class is given as an argument `(some_enumerable.my_all?(String)).`

- Your method doesn't consider the possibility when a Regex is given as an argument `(some_enumerable.my_all?(/w/)).`

- Your method doesn't consider the possibility when a number is given as an argument `(some_enumerable.my_all?(5)).`